src/WebJobs.Script.WebHost/Controllers/InstanceController.cs

```diff
@@ -41,22 +41,27 @@ public async Task<IActionResult> Assign([FromBody] EncryptedHostAssignmentContex
 
             var assignmentContext = _startupContextProvider.SetContext(encryptedAssignmentContext);
 
-            // before starting the assignment we want to perform as much
-            // up front validation on the context as possible
-            string error = await _instanceManager.ValidateContext(assignmentContext);
-            if (error != null)
-            {
-                return StatusCode(StatusCodes.Status400BadRequest, error);
-            }
+            string error;
+            
+            // Skip MSI Specialization and validation for Legion
+            if (!_environment.IsLinuxConsumptionOnLegion()) {
```

Comment:
we should get the correct instancemanager type injected by DI. (no explicit checks here)

Reply:
So currently, the LegionInstanceManager will be injected already, and the ValidateContext() just throws a NotImplementedException. Should I just get rid of this explicit check and just have the ValidateContext() as no-op?